libs/orders/src/lib/order-hooks/use-order-validation.spec.tsx

```diff
@@ -90,107 +91,148 @@ function setup(
-  ${MarketState.Proposed}          | ${ERROR.MARKET_WAITING}
-`(
-  'Returns an error message for "$marketState" market',
-  async ({ state, errorMessage }) => {
-    const { result } = setup({ market: { ...defaultOrder.market, state } });
-    expect(result.current).toEqual(errorMessage);
-  }
-);
-
-it.each`
-  tradingMode                            | errorMessage
-  ${MarketTradingMode.BatchAuction}      | ${ERROR.MARKET_CONTINUOUS_LIMIT}
-  ${MarketTradingMode.MonitoringAuction} | ${ERROR.MARKET_CONTINUOUS_LIMIT}
-  ${MarketTradingMode.OpeningAuction}    | ${ERROR.MARKET_CONTINUOUS_LIMIT}
-`(
-  'Returns an error message when trying to submit a non-limit order for a "$tradingMode" market',
-  async ({ tradingMode, errorMessage }) => {
-    const { result } = setup({
-      market: { ...defaultOrder.market, tradingMode },
-      orderType: VegaWalletOrderType.Market,
+  it('Returns an error message when the keypair is tainted', () => {
+    const { result } = setup(defaultOrder, {
+      keypair: { ...defaultWalletContext.keypair, tainted: true },
     });
-    expect(result.current).toEqual(errorMessage);
-  }
-);
-
-it.each`
-  tradingMode                            | orderTimeInForce                  | errorMessage
-  ${MarketTradingMode.BatchAuction}      | ${VegaWalletOrderTimeInForce.FOK} | ${ERROR.MARKET_CONTINUOUS_TIF}
-  ${MarketTradingMode.MonitoringAuction} | ${VegaWalletOrderTimeInForce.FOK} | ${ERROR.MARKET_CONTINUOUS_TIF}
-  ${MarketTradingMode.OpeningAuction}    | ${VegaWalletOrderTimeInForce.FOK} | ${ERROR.MARKET_CONTINUOUS_TIF}
-  ${MarketTradingMode.BatchAuction}      | ${VegaWalletOrderTimeInForce.IOC} | ${ERROR.MARKET_CONTINUOUS_TIF}
-  ${MarketTradingMode.MonitoringAuction} | ${VegaWalletOrderTimeInForce.IOC} | ${ERROR.MARKET_CONTINUOUS_TIF}
-  ${MarketTradingMode.OpeningAuction}    | ${VegaWalletOrderTimeInForce.IOC} | ${ERROR.MARKET_CONTINUOUS_TIF}
-  ${MarketTradingMode.BatchAuction}      | ${VegaWalletOrderTimeInForce.GFN} | ${ERROR.MARKET_CONTINUOUS_TIF}
-  ${MarketTradingMode.MonitoringAuction} | ${VegaWalletOrderTimeInForce.GFN} | ${ERROR.MARKET_CONTINUOUS_TIF}
-  ${MarketTradingMode.OpeningAuction}    | ${VegaWalletOrderTimeInForce.GFN} | ${ERROR.MARKET_CONTINUOUS_TIF}
-`(
-  'Returns an error message when submitting a limit order with a "$orderTimeInForce" value to a "$tradingMode" market',
-  async ({ tradingMode, orderTimeInForce, errorMessage }) => {
+    expect(result.current).toStrictEqual({ isDisabled: false, message: `` });
+  });
+
+  it.each`
+    state
+    ${MarketState.Settled}
+    ${MarketState.Rejected}
+    ${MarketState.TradingTerminated}
+  `(
+    'Returns an error message for market state when no longer accepting orders',
+    ({ state }) => {
+      const { result } = setup({ market: { ...defaultOrder.market, state } });
+      expect(result.current).toStrictEqual({
+        isDisabled: true,
+        message: `This market is ${marketTranslations(
```

Comment:
If I'm reading this correctly, then I believe it should say "This market is (state) and not accepting orders", because, for example, a market that is pending will not have been able to take orders yet, ever.